lib/msf/core/auxiliary/command_shell.rb

```diff
@@ -51,6 +51,8 @@ def start_session(obj, info, ds_merge, crlf = false, sock = nil)
     framework.sessions.register(sess)
     sess.process_autoruns(datastore)
 
+    report_session(:session => sess)
+
```

Comment:
```suggestion
    # Notify the framework that we have a new session opening up...
    # Don't let errant event handlers kill our session
    begin
      framework.events.on_session_open(sess)
    rescue ::Exception => e
      wlog("Exception in on_session_open event handler: #{e.class}: #{e}")
      wlog("Call Stack\n#{e.backtrace.join("\n")}")
    end
```
Instead of implementing `report_session` in `core/auxiliary/report.rb` this should likely call `framework.events.on_session_open(session)` see: https://github.com/rapid7/metasploit-framework/blob/6ecaeaf7d7cdf410517824f757bb531e0735d782/lib/msf/core/handler.rb#L280-L287
This will allow framework mechanisms that watch for session events to also get notified of the session.